python/ray/autoscaler/_private/autoscaler.py

```diff
@@ -71,6 +70,7 @@ def __init__(self,
         # Keep this before self.reset (self.provider needs to be created
         # exactly once).
         self.provider = None
+        self.resource_demand_scheduler = None
```

Comment:
I do not understand.

Reply:
We cannot put this in self.reset because it is called everytime we do an autoscaler update and thus destroying the `self.resource_demand_scheduler` instance.